code/datums/wires/canister.dm

```diff
@@ -0,0 +1,33 @@
+/datum/wires/canister
+	holder_type = /obj/machinery/portable_atmospherics/canister
+	proper_name = "Canister"
+
+/datum/wires/canister/New(atom/holder)
+	wires = list(WIRE_VALVE, WIRE_SHIELDING, WIRE_REGULATOR_MIN, WIRE_REGULATOR_MAX, WIRE_TANK_EJECT, WIRE_REACTION_SUPPRESSION)
+	..()
+
+/datum/wires/canister/on_pulse(wire)
+	var/obj/machinery/portable_atmospherics/canister/the_canister = holder
+	if(!the_canister.internal_cell)
+		return
+	switch(wire)
+		if(WIRE_VALVE)
+			the_canister.toggle_valve(usr)
```

Comment:
All these procs and proccalls should have a second argument so we know if they're set off by a person or a rig, so that we know whether to log the user or rigger and how to word the log message.